warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -286,29 +284,26 @@ public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> op
     
     // this method will prune any ivarator cache directories that do not have a valid configuration.
     private void pruneIvaratorCacheDirs() throws InterruptedIOException {
-        if (ivaratorCacheDirConfigs.isEmpty()) {
-            return;
-        }
-        IvaratorCacheDirConfig validConfig = null;
+        List<IvaratorCacheDirConfig> configsToRemove = new ArrayList<>();
         for (IvaratorCacheDirConfig config : ivaratorCacheDirConfigs) {
-            if (hasValidBasePath(config)) {
-                validConfig = config;
-                break;
+            if (!hasValidBasePath(config)) {
+                configsToRemove.add(config);
             }
         }
-        if (validConfig != null) {
-            ivaratorCacheDirConfigs = Collections.singletonList(validConfig);
-        } else {
-            ivaratorCacheDirConfigs = Collections.EMPTY_LIST;
-        }
+        ivaratorCacheDirConfigs.removeAll(configsToRemove);
     }
     
     private boolean hasValidBasePath(IvaratorCacheDirConfig config) throws InterruptedIOException {
         if (config.isValid()) {
             try {
                 Path basePath = new Path(config.getBasePathURI());
                 FileSystem fileSystem = this.getFileSystemCache().getFileSystem(basePath.toUri());
-                return isWritablePath(basePath, fileSystem);
+                
+                // Note: The ivarator config base paths are used by ALL queries which run on the system, so there
+                // should be no harm in creating these directories if they do not already exist by this point.
+                // Also, since we are selecting these directories intentionally for use by the ivarators, it
+                // should be a given that we have write permissions.
+                return fileSystem.exists(basePath) || fileSystem.mkdirs(basePath);
```

Comment:
I think the real problem may have been the cleanup of said directories.  We were creating way too many and subsequently if an ivarator directory was to be reused (iterator torn down and rebuilt) then it was unable to find the files from the previous iteration.  We need to find the other ticket that resulted in this change.

Reply:
@ivakegg Not sure if this is the other ticket you wanted to find, but just in case, work was done in PR #1125 for to address lazily creating ivarator cache dirs to resolve issue #1059.

Reply:
@ivakegg That's not what the ticket associated with any of the changes indicated.  If there's another one I'm missing, can you point me to it?

Reply:
I will have to take a look tomorrow.   I am probably remembering incorrectly, but I need to verify.  Sorry for the delay.

Reply:
That's fine.  if it turns out that there is a legitimate concern there, then I would prefer to take the time to update the code to handle that with multiple ivarator cache dirs available instead of just limiting the query iterator to one.

Reply:
For performance reasons, I am ok with reducing this check as you have done.